zio-http/src/main/scala/zio/http/Version.scala

```diff
@@ -17,24 +17,40 @@
 package zio.http
 
 sealed trait Version { self =>
-  def ++(that: Version): Version = (self, that) match {
-    case (Version.Http_1_0, Version.Http_1_0) => Version.Http_1_0
-    case _                                    => Version.Http_1_1
-  }
+
+  /**
+   * An integer representing the version. The integer is arbitrary, it is only
+   * guaranteed that later versions of the HTTP protocol have a higher number,
+   * and that the "default" version has the lowest number.
+   */
+  def ordinal: Int
+
+  def ++(that: Version): Version =
+    if (self.ordinal > that.ordinal) self else that
 
   def combine(that: Version): Version = self ++ that
 
   def isHttp1_0: Boolean = self == Version.Http_1_0
 
   def isHttp1_1: Boolean = self == Version.Http_1_1
-
 }
 
 object Version {
   val `HTTP/1.0`: Version = Http_1_0
   val `HTTP/1.1`: Version = Http_1_1
 
-  case object Http_1_0 extends Version
+  /**
+   * Indicates no preference for version. The default version will be used.
+   */
+  case object Default extends Version {
```

Comment:
Why implement it as a `case object` you'll have to deal with in your pattern matchings and not as `def Default = Http_1_1`?